### PR TITLE
Added support for decimals on beanCoin usage

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/com/mazawrath/beanbot/commands/beanmarket/BeanMarketCommand.java [hunk 1]
```diff
@@ -53,12 +54,12 @@ public void onCommand(String[] args, ServerTextChannel serverTextChannel) {
...
                             .setThumbnail(obj.getString("Logo"))
-                            .addInlineField("Price", obj.getBigDecimal("Price").setScale(2, RoundingMode.HALF_UP).toString())
-                            .addInlineField("Daily Percentage Change", obj.getBigDecimal("Percentage Change")  + "%")
-                            .addInlineField("Previously Closed At", obj.getBigDecimal("Previous Close").setScale(2, RoundingMode.HALF_UP).toString())
-                            .addInlineField("50 Day Percentage Change", obj.getBigDecimal("50 Day Percentage Change").setScale(2, RoundingMode.HALF_UP).toString()  + "%")
-                            .addInlineField("Yearly High", obj.getBigDecimal("Year High").setScale(2, RoundingMode.HALF_UP).toString())
-                            .addInlineField("Yearly Low", obj.getBigDecimal("Year Low").setScale(2, RoundingMode.HALF_UP).toString());
+                            .addInlineField("Price", Points.pointsToString(obj.getBigDecimal("Price").setScale(2, RoundingMode.HALF_UP)))
+                            .addInlineField("Daily Percentage Change", Points.pointsToString(obj.getBigDecimal("Percentage Change"))  + "%")
```
Comment (Mazawrath) [inline src/main/java/com/mazawrath/beanbot/commands/beanmarket/BeanMarketCommand.java]: This is not meant to have a `ß`

File: src/main/java/com/mazawrath/beanbot/commands/beanmarket/BeanMarketCommand.java [hunk 2]
```diff
@@ -53,12 +54,12 @@ public void onCommand(String[] args, ServerTextChannel serverTextChannel) {
...
+                            .addInlineField("Previously Closed At", Points.pointsToString(obj.getBigDecimal("Previous Close").setScale(2, RoundingMode.HALF_UP)))
+                            .addInlineField("50 Day Percentage Change", Points.pointsToString(obj.getBigDecimal("50 Day Percentage Change").setScale(2, RoundingMode.HALF_UP)) + "%")
```
Comment (Mazawrath) [inline src/main/java/com/mazawrath/beanbot/commands/beanmarket/BeanMarketCommand.java]: This is not meant to have a `ß`

File: src/main/java/com/mazawrath/beanbot/commands/UserInfoCommand.java [hunk 3]
```diff
@@ -81,7 +81,7 @@ public void onCommand(String[] args, DiscordApi api, ServerTextChannel serverTex
                     .addInlineField("Username", user.getDiscriminatedName())
                     .addInlineField("Roles", roleList[0])
                     .addInlineField("Date Joined", formatted)
-                    .addInlineField("beanCoin Balance", String.valueOf(points.getBalance(user.getIdAsString(), server.getIdAsString())) + " beanCoin")
+                    .addInlineField("beanCoin Balance", Points.pointsToString(points.getBalance(user.getIdAsString(), server.getIdAsString())) + " beanCoin")
```
Comment (Mazawrath) [inline src/main/java/com/mazawrath/beanbot/commands/UserInfoCommand.java]:
```suggestion
                    .addInlineField("beanCoin Balance", Points.pointsToString(points.getBalance(user.getIdAsString(), server.getIdAsString())))
```

Comment (Mazawrath) [reply src/main/java/com/mazawrath/beanbot/commands/UserInfoCommand.java]: Either that or in this one case it could be without the `ß`

**Review by Mazawrath — CHANGES_REQUESTED** [review]

**Review by Mazawrath — APPROVED** [review]

**Comment by ghost** [comment]
To update previous values in DB from number to decimal with the format proposed, needs to execute this command on RethinkDB admin interface (replace TABLE_ID with the right table):

```sql
r.db('beanBotPoints').table('TABLE_ID').forEach(function (row){
  return r.db('beanBotPoints').table('TABLE_ID').filter({"id": row.getField('id')}).update({"Points": r.expr("P_").add(row.getField('Points').coerceTo('string'), ".00")})
})
```

**Pull request merged**